rir/src/compiler/opt/constantfold.cpp

```diff
@@ -65,69 +76,85 @@ void Constantfold::apply(RirCompiler& cmp, Closure* function,
             auto next = ip + 1;
 
             // Constantfolding of some common operations
-            FOLD(Add, symbol::Add);
-            FOLD(Sub, symbol::Sub);
-            FOLD(Mul, symbol::Mul);
-            FOLD(Div, symbol::Div);
-            FOLD(IDiv, symbol::Idiv);
-            FOLD(Lt, symbol::Lt);
-            FOLD(Gt, symbol::Gt);
-            FOLD(Lte, symbol::Le);
-            FOLD(Gte, symbol::Ge);
-            FOLD(Eq, symbol::Eq);
-            FOLD(Neq, symbol::Ne);
-            FOLD(Pow, symbol::Pow);
-
-            if (auto assume = Assume::Cast(i)) {
-                auto condition = assume->arg<0>().val();
-                if (auto isObj = IsObject::Cast(condition))
-                    if (!isObj->arg<0>().val()->type.maybeObj())
+            FOLD2_SIMPLE(Add, symbol::Add);
+            FOLD2_SIMPLE(Sub, symbol::Sub);
+            FOLD2_SIMPLE(Mul, symbol::Mul);
+            FOLD2_SIMPLE(Div, symbol::Div);
+            FOLD2_SIMPLE(IDiv, symbol::Idiv);
+            FOLD2_SIMPLE(Lt, symbol::Lt);
+            FOLD2_SIMPLE(Gt, symbol::Gt);
+            FOLD2_SIMPLE(Lte, symbol::Le);
+            FOLD2_SIMPLE(Gte, symbol::Ge);
+            FOLD2_SIMPLE(Eq, symbol::Eq);
+            FOLD2_SIMPLE(Neq, symbol::Ne);
+            FOLD2_SIMPLE(Pow, symbol::Pow);
+
+            auto convertsToLglWithoutWarning = [](SEXP arg) {
```

Comment:
The name is already too long. So logical(I prefer boolean) instead would not hurt here.

Reply:
in R parlance logical is a logical vector, whereas boolean a *scalar* logical vector.